libkodimote/libraryitem.cpp

```diff
@@ -490,8 +490,14 @@ QString LibraryItem::year() const
 
 void LibraryItem::setYear(const QString &year)
 {
-    m_year = year;
-    emit yearChanged();
+    if (year == "0") {
+        //can't change year, as it's const
```

Comment:
Because it would be less code, easier to read, and easier to maintain (I actually should have put the emit below the if/else)